### PR TITLE
Adds redirects and updates links for learn.hashicorp

### PR DESCRIPTION
**Pull request by topfunky**
This PR modifies Fastly redirects and other URLs throughout the documentation to point to Consul resources at https://learn.hashicorp.com/

### CODE REVIEW

File: website/source/segmentation.html.erb [hunk 1]
```diff
@@ -194,13 +194,13 @@ Secure Sockets Layer
   <section class='g-section g-cta-section'>
     <div>
       <h2>Ready to get started?</h2>
-      <a href="/intro/getting-started/connect.html" class="g-btn white download">
+      <a href="/downloads.html" class="g-btn white download">
```
Comment (pearkes) [inline website/source/segmentation.html.erb]: Ha nice.

File: website/source/intro/getting-started/agent.html.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ description: >
   as data loss is inevitable in a failure scenario.
 ---
 
+**Note: This guide is temporarily being managed in a different repo. Please contact kait@hashicorp.com if you have any questions.
```
Comment (pearkes) [inline website/source/intro/getting-started/agent.html.md]: I actually think we can remove these. I could see people missing them if they went to update the guide. Prefer to have @kaitlincarter-hc just watch PRs closely.

**Comment by pearkes** [comment]
Want to confirm for any future PR readers, though this documentation will stay here for now, we will eventually have the source of learn.hashicorp.com docs be within this repo, correct @topfunky / @kaitlincarter-hc?

File: website/source/layouts/intro.erb [hunk 3]
```diff
@@ -41,37 +41,8 @@
...
-            <a href="/intro/getting-started/kv.html">KV Data</a>
-          </li>
-          <li<%= sidebar_current("gettingstarted-ui") %>>
-            <a href="/intro/getting-started/ui.html">Web UI</a>
-          </li>
-          <li<%= sidebar_current("gettingstarted-nextsteps") %>>
-            <a href="/intro/getting-started/next-steps.html">Next Steps</a>
-          </li>
-        </ul>
+      <li<%= sidebar_current("starte") %>>
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/layouts/intro.erb]: Nope! Typo.

File: website/redirects.txt [hunk 4]
```diff
@@ -38,3 +38,14 @@
...
+/intro/getting-started/install.html      https://learn.hashicorp.com/consul/getting-started/install.html
+/intro/getting-started/agent.html        https://learn.hashicorp.com/consul/getting-started/agent.html
+/intro/getting-started/services.html     https://learn.hashicorp.com/consul/getting-started/services.html
+/intro/getting-started/connect.html      https://learn.hashicorp.com/consul/getting-started/connect.html
+/intro/getting-started/join.html         https://learn.hashicorp.com/consul/getting-started/join.html
+/intro/getting-started/checks.html       https://learn.hashicorp.com/consul/getting-started/checks.html
+/intro/getting-started/kv.html           https://learn.hashicorp.com/consul/getting-started/kv.html
+/intro/getting-started/ui.html           https://learn.hashicorp.com/consul/getting-started/ui.html
+/intro/getting-started/next-steps.html   https://learn.hashicorp.com/consul/getting-started/next-steps.html
+/intro/getting-started/consul.d/web.json https://learn.hashicorp.com/consul/getting-started/consul.d/web.json
```
Comment (pearkes) [inline website/redirects.txt]: This consul.d and web.json file should just be removed, and we shouldn't maintain redirects for it. Seems like it was just a mistake.

https://github.com/hashicorp/consul/commit/9002bfa37df3aa39024030cb8c312c94cca7c926#diff-a970970e9f0143c37300d2d6ea25d501

File: website/source/intro/getting-started/agent.html.md (hunk 2 above)
Comment (topfunky) [reply website/source/intro/getting-started/agent.html.md]: Files have been left in the repo but the prose has been removed.

**Comment by topfunky** [comment]
> Want to confirm for any future PR readers, though this documentation will stay here for now, we will eventually have the source of learn.hashicorp.com docs be within this repo, correct @topfunky / @kaitlincarter-hc?

Yes, the source for all guides and getting started guides will be moved to this `consul` repo in order to make contributions easier.

File: website/redirects.txt (hunk 4 above)
Comment (topfunky) [reply website/redirects.txt]: The redirect to the demo file has been removed as requested. 8a940c69a60

**Pull request merged**